### PR TITLE
Clean up remaining ESLint errors/warnings

### LINKED ISSUE #40: Clean up ESLint errors
(continued from #39 conversation)

``` sh
$$ npm run lint

> mozilla-universal-search-addon@1.0.1 lint /Users/pdehaan/dev/github/universal-search-addon
> eslint .


bootstrap.js
   5:17  error  Cc is defined but never used         no-unused-vars
   5:33  error  Ci is defined but never used         no-unused-vars
   5:57  error  Cm is defined but never used         no-unused-vars
  11:9   error  startup is defined but never used    no-unused-vars
  16:9   error  shutdown is defined but never used   no-unused-vars
  32:9   error  install is defined but never used    no-unused-vars
  38:9   error  uninstall is defined but never used  no-unused-vars

lib/main.js
    7:17  error  Cc is defined but never used                no-unused-vars
    7:33  error  Ci is defined but never used                no-unused-vars
    7:57  error  Cm is defined but never used                no-unused-vars
   18:4   error  EXPORTED_SYMBOLS is defined but never used  no-unused-vars
   86:6   error  document is defined but never used          no-unused-vars
  143:4   error  Main is defined but never used              no-unused-vars

lib/ui/GoButton.js
  5:4  error  EXPORTED_SYMBOLS is defined but never used  no-unused-vars

lib/ui/Popup.js
   14:4  error  EXPORTED_SYMBOLS is defined but never used  no-unused-vars
  257:4  error  Expected no return value                    consistent-return

lib/ui/Urlbar.js
  7:4  error  EXPORTED_SYMBOLS is defined but never used  no-unused-vars

✖ 17 problems (17 errors, 0 warnings)
```

I can probably clean up most of these by just ignoring the `no-unused-vars` rule on specific lines without needing to disable the rule globally.

I'll try and come up with a better explanation of the `consistent-return` issue while I'm at it.

### PR DESCRIPTION
**Pull request by pdehaan**
Fixes #40 

r? @6a68

### CODE REVIEW

**Comment by pdehaan** [comment]
Current status, zero errors/warnings:

``` sh
$ npm run lint

> mozilla-universal-search-addon@1.0.1 lint /Users/pdehaan/dev/github/universal-search-addon
> eslint .

$ echo $?
0
```

File: bootstrap.js [hunk 1]
```diff
@@ -2,18 +2,18 @@
...
 
-const { classes: Cc, interfaces: Ci, utils: Cu, manager: Cm } = Components;
+const { utils: Cu } = Components;
 
 Cu.import('resource://gre/modules/XPCOMUtils.jsm');
 XPCOMUtils.defineLazyModuleGetter(this, 'Main',
   'chrome://universalsearch-lib/content/main.js');
 
-function startup(data, reason) {
+function startup(data, reason) { // eslint-disable-line no-unused-vars
```
Comment (pdehaan) [inline bootstrap.js:11]: Basically lets just ignore the `no-unused-vars` for these specific functions.

File: lib/main.js [hunk 2]
```diff
@@ -83,7 +84,6 @@ var loadIntoWindow = function(win) {
 var unloadFromWindow = function(win) {
   console.log('unloadFromWindow start');
 
-  var document = win.document;
```
Comment (pdehaan) [inline lib/main.js:86]: Deleting, per https://github.com/mozilla/universal-search-addon/pull/39#issuecomment-119779011

File: .eslintrc [hunk 3]
```diff
@@ -1,4 +1,5 @@
 extends: universal-search
 
 rules:
+  consistent-return: 0
```
Comment (pdehaan) [inline .eslintrc:4]: > `consistent-return` - require `return` statements to either always or never specify values
> &mdash; via http://eslint.org/docs/rules/consistent-return

**Comment by pdehaan** [comment]
Rebased.

File: .gitignore [hunk 4]
```diff
@@ -1,2 +1,3 @@
 node_modules
 dist
+*.log
```
Comment (jaredhirsch) [inline .gitignore:3]: what's this for?

File: lib/ui/Popup.js [hunk 5]
```diff
@@ -199,13 +202,13 @@ Popup.prototype = {
...
-          url: Components.classes["@mozilla.org/intl/texttosuburi;1"].
+          url: Components.classes['@mozilla.org/intl/texttosuburi;1'].
                 getService(Components.interfaces.nsITextToSubURI).
-                unEscapeURIForUI("UTF-8", controller.getValueAt(i)),
+                unEscapeURIForUI('UTF-8', controller.getValueAt(i)),
           image: imgMatches ? imgMatches[1] : null,
           title: controller.getCommentAt(i),
           type: controller.getStyleAt(i),
-          text: controller.searchString.replace(/^\s+/, "").replace(/\s+$/, "")
+          text: controller.searchString.trim()
```
Comment (jaredhirsch) [inline lib/ui/Popup.js:211]: :beers:

File: .gitignore (hunk 4 above)
Comment (jaredhirsch) [inline .gitignore:3]: meh, :ship:

Comment (pdehaan) [inline .gitignore:3]: So I stop trying to merge in my npm error *.log files whenever npm vomits all over the place.

**Comment by jaredhirsch** [comment]
looks great, @pdehaan! Thanks so much for the contribution

**Pull request merged**